codegen/endpoint_registry.go

```diff
@@ -1,10 +1,104 @@
 package codegen
 
+import "github.com/hashicorp/vault/sdk/framework"
+
 // endpointRegistry is a registry of all the endpoints we'd
 // like to have generated, along with the type of template
 // we should use.
 // IMPORTANT NOTE: To support high quality, only add one
 // endpoint per PR.
-var endpointRegistry = map[string]templateType{
-	"/transform/role/{name}": templateTypeResource,
+var endpointRegistry = map[string]*additionalInfo{
+	"/transform/alphabet/{name}": {
+		TemplateType: templateTypeResource,
+	},
+	"/transform/decode/{role_name}": {
+		TemplateType: templateTypeDataSource,
+		AdditionalParameters: []*templatableParam{
+			{
+				OASParameter: &framework.OASParameter{
+					Name:        "decoded_value",
+					Description: "The result of decoding a value.",
+					Schema: &framework.OASSchema{
+						Type:         "string",
+						DisplayAttrs: &framework.DisplayAttributes{},
+					},
+				},
+				Computed: true,
+			},
+			{
+				OASParameter: &framework.OASParameter{
+					Name:        "batch_results",
+					Description: "The result of decoding batch_input.",
+					Schema: &framework.OASSchema{
+						Type: "array",
+						Items: &framework.OASSchema{
+							Type: "object",
+						},
+						DisplayAttrs: &framework.DisplayAttributes{},
+					},
+				},
+				Computed: true,
+			},
+		},
+	},
+	"/transform/encode/{role_name}": {
+		TemplateType: templateTypeDataSource,
+		AdditionalParameters: []*templatableParam{
+			{
+				OASParameter: &framework.OASParameter{
+					Name:        "encoded_value",
+					Description: "The result of encoding a value.",
+					Schema: &framework.OASSchema{
+						Type:         "string",
+						DisplayAttrs: &framework.DisplayAttributes{},
+					},
+				},
+				Computed: true,
+			},
+			{
+				OASParameter: &framework.OASParameter{
+					Name:        "batch_results",
+					Description: "The result of encoding batch_input.",
+					Schema: &framework.OASSchema{
+						Type: "array",
+						Items: &framework.OASSchema{
+							Type: "object",
+						},
+						DisplayAttrs: &framework.DisplayAttributes{},
+					},
+				},
+				Computed: true,
+			},
+		},
+	},
+	"/transform/role/{name}": {
+		TemplateType: templateTypeResource,
+	},
+	"/transform/template/{name}": {
+		TemplateType: templateTypeResource,
+	},
+	"/transform/transformation/{name}": {
+		TemplateType: templateTypeResource,
+		AdditionalParameters: []*templatableParam{
+			{
+				OASParameter: &framework.OASParameter{
+					Name:        "templates",
+					Description: "Templates configured for transformation.",
+					Schema: &framework.OASSchema{
+						Type: "array",
+						Items: &framework.OASSchema{
+							Type: "string",
+						},
+						DisplayAttrs: &framework.DisplayAttributes{},
+					},
+				},
+				Computed: true,
+			},
+		},
+	},
+}
+
+type additionalInfo struct {
+	TemplateType         templateType
+	AdditionalParameters []*templatableParam
```

Comment:
Hm, this would be a departure from the coding approach that's been taken by the team so far. I'm hesitant to do it because I don't want folks looking at it later and wondering what I was thinking. Rather than changing this here, can we instead take it into a team meeting for discussion before changing it?

Reply:
There wasn't a finish on our discussion, but I don't want to hold this up. I think this aligns with the [Pass Values](https://github.com/golang/go/wiki/CodeReviewComments#pass-values) principle, though it's not exactly the same, so I'll go ahead and update it.

Reply:
This is done in https://github.com/terraform-providers/terraform-provider-vault/pull/749/commits/f676d9f7cd401bd3f31eb32e991f8c4d6bbf6046